### ISSUE TITLE
Code completion not working

### ISSUE THREAD

**Issue by maschere** [benfry/processing4#177]
## Description

Code completion window does not show after pressing ctrl + space. It works fine in Processing 3.5.4

## Expected Behavior

Code completion window pops up after pressing ctrl + space

## Steps to Reproduce

1. open blank project
2. enable "code completion with ctrl+space" in preferences
3.  type **int testvar = 1;**
4. type **testv** in the next line and then press ctrl+space

## Your Environment

 

* Processing version: 4.0a3
* Operating System and OS version: Windows 10 x64

**Comment by benfry** [comment]
Confirmed to be broken in the current source.

@sampottinger it looks like this was broken back in alpha 1, did you have it working properly?

**Comment by benfry** [comment]
Tracked down to 51145a88862840ce6ecb1220714335686c02515e, though unfortunately that's the monolithic commit for the JDK support, ANTLR updates, etc, so it's still hard to know if this is a large problem or just something overlooked. 

The internal issue is that `javaCode` is not being set when the code cannot compile, which for code completion is always the case. So there's a `NullPointerException` when that's accessed, and no completions are available. Hoping that there's a simple fix to get the incomplete code assigned to the variable, and we can get this working again.

**Comment by sampottinger** [comment]
Hey there! It appears that antlr gives up before it can apply the text transformations. I have a branch over at #219 that demonstrates allowing the the text transformation process to continue. That said, antlr does appear to throw out a lot before it is able to recover from a syntax error and my demo only works with the semicolon included at the end. Still working on that.

<img width="768" alt="Screen Shot 2021-07-03 at 3 39 21 PM" src="https://user-images.githubusercontent.com/110391/124369052-695f9980-dc1c-11eb-9609-d781a218a7db.png">

**Comment by sampottinger** [comment]
Hey there! Ah dang... right after I sent that @benfry, I realized there was another place we could be dropping the tree and, indeed, fixing that brought back autocomplete. See #219.

@maschere, thank you for this report (and your patience). I have used your test case in the attached example.

![Screen Shot 2021-07-03 at 4 50 42 PM](https://user-images.githubusercontent.com/110391/124369276-edb31c00-dc1e-11eb-8a7d-cd1d2257b0fe.png)
